core/src/main/java/com/linecorp/armeria/client/RefreshingAddressResolverGroup.java

```diff
@@ -137,16 +140,19 @@
     @Override
     public void close() {
         super.close();
-        while (!cache.isEmpty()) {
-            for (final Iterator<Entry<String, CompletableFuture<CacheEntry>>> i = cache.entrySet().iterator();
-                 i.hasNext();) {
-                final Entry<String, CompletableFuture<CacheEntry>> entry = i.next();
-                i.remove();
-                entry.getValue().handle((cacheEntry, cause) -> {
+        cache.invalidateAll();
+    }
+
+    private static Cache<String, CompletableFuture<CacheEntry>> buildCache(String cacheSpec) {
+        final Caffeine<Object, Object> b = Caffeine.from(cacheSpec);
```

Comment:
What happens if the spec is off?

Reply:
I decided to keep the DNS cache as nonnull, so I've removed `off` from the spec.

Reply:
Yeah, that's a good idea. 👍